### PR TITLE
[RBAC] BREAKING CHANGE: az role assignment delete: always prompt for confirmation if not specify --yes

### PR DESCRIPTION
**Pull request by arrownj**
**Description**  
Previously if user specify argument other than --yes in `az role assignment delete`, CLI will not prompt for confirmation. This is wired and make user confused.

This update is to update this behavior and always prompt for confirmation for this `delete` command.

**Testing Guide**  

**History Notes**  

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.  
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

File: src/azure-cli/azure/cli/command_modules/role/custom.py [hunk 1]
```diff
@@ -487,9 +487,12 @@ def delete_role_assignments(cmd, ids=None, assignee=None, role=None, resource_gr
         for i in ids:
             assignments_client.delete_by_id(i)
         return
-    if not any([ids, assignee, role, resource_group_name, scope, assignee, yes]):
+    if not yes:
         from knack.prompting import prompt_y_n
-        msg = 'This will delete all role assignments under the subscription. Are you sure?'
+        msg = 'This will delete the matched role assignments. Are you sure to continue?'
+        if not any([ids, assignee, role, resource_group_name, scope, assignee]):
+            msg = 'WARNING: As no parameter is specified, all role assignments under the subscription will be '\
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/role/custom.py:494]: `WARNING: ` will be added by Knack automatically if color is disabled.

**Comment by yonzhan** [comment]
RBAC

**Comment by jiasli** [comment]
Related to https://github.com/Azure/azure-cli/issues/13577

**Comment by yungezz** [comment]
hi @arrownj what's plan of this PR?

**Comment by yungezz** [comment]
hi @arrownj could you pls share plan on this PR? thanks.

**Comment by arrownj** [comment]
> hi @arrownj Xiaojian Xu FTE could you pls share plan on this PR? thanks.

Hi @yungezz , this long term solution is transferred to Jiashuo in https://github.com/Azure/azure-cli/issues/13577. I have added my PR link in the issue for reference. And I will close this PR first.

**Pull request closed without merging**